### ISSUE TITLE
How to change extra-data files

### ISSUE THREAD

**Issue by FinMan28** [liren2515/DrapeNet#19]
Hello, can you please tell me where the files body_info_f.npz and body_info_npz in extra-data come from? How are they produced and how to change their content? I would like to change pose and betas of the model.

**Comment by liren2515** [comment]
If you mean changing the pose and betas of the body, you can get them from any SMPL related body mesh dataset, such as 3DPW. Here is a survey for it https://github.com/tinatiansjz/hmr-survey?tab=readme-ov-file#datasets. You don't need the generate the data as the one in body_info_f.npz.

Since the `tfs_c_inv ` we used here is to align the rest pose of CLOTH3D to the rest pose of SMPL. 
```
data_body = np.load(os.path.join(extra_dir, 'body_info_f.npz'))
tfs_c_inv = torch.FloatTensor(data_body['tfs_c_inv']).to(device)
```
The rest pose of CLOTH3D is defined as below instead of a zero vector as that defined in SMPL.
```
pose = pose.reshape(1, 24, 3)
pose[0, 1, 2] = 0.15
pose[0, 2, 2] = -0.15
pose = pose.reshape(1, 72)
```

**Comment by Yashaswini2039** [comment]
> If you mean changing the pose and betas of the body, you can get them from any SMPL related body mesh dataset, such as 3DPW. Here is a survey for it https://github.com/tinatiansjz/hmr-survey?tab=readme-ov-file#datasets. You don't need the generate the data as the one in body_info_f.npz.
> 
> Since the `tfs_c_inv ` we used here is to align the rest pose of CLOTH3D to the rest pose of SMPL.
> 
> ```
> data_body = np.load(os.path.join(extra_dir, 'body_info_f.npz'))
> tfs_c_inv = torch.FloatTensor(data_body['tfs_c_inv']).to(device)
> ```
> 
> The rest pose of CLOTH3D is defined as below instead of a zero vector as that defined in SMPL.
> 
> ```
> pose = pose.reshape(1, 24, 3)
> pose[0, 1, 2] = 0.15
> pose[0, 2, 2] = -0.15
> pose = pose.reshape(1, 72)
> ```

I have an .obj file containing posed SMPL, how can i use that for draping.
As i checked it out u have taken the pose component from sample-pose.pt file and i wanned to know can i extract that pose parameters from my posed_smpl.obj file

**Comment by liren2515** [comment]
> > If you mean changing the pose and betas of the body, you can get them from any SMPL related body mesh dataset, such as 3DPW. Here is a survey for it https://github.com/tinatiansjz/hmr-survey?tab=readme-ov-file#datasets. You don't need the generate the data as the one in body_info_f.npz.
> > Since the `tfs_c_inv ` we used here is to align the rest pose of CLOTH3D to the rest pose of SMPL.
> > ```
> > data_body = np.load(os.path.join(extra_dir, 'body_info_f.npz'))
> > tfs_c_inv = torch.FloatTensor(data_body['tfs_c_inv']).to(device)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The rest pose of CLOTH3D is defined as below instead of a zero vector as that defined in SMPL.
> > ```
> > pose = pose.reshape(1, 24, 3)
> > pose[0, 1, 2] = 0.15
> > pose[0, 2, 2] = -0.15
> > pose = pose.reshape(1, 72)
> > ```
> 
> I have an .obj file containing posed SMPL, how can i use that for draping. As i checked it out u have taken the pose component from sample-pose.pt file and i wanned to know can i extract that pose parameters from my posed_smpl.obj file

Yes, you can. You need to fit smpl model to your obj mesh to get the corresponding \theta and \beta.

**Comment by Yashaswini2039** [comment]
> > > If you mean changing the pose and betas of the body, you can get them from any SMPL related body mesh dataset, such as 3DPW. Here is a survey for it https://github.com/tinatiansjz/hmr-survey?tab=readme-ov-file#datasets. You don't need the generate the data as the one in body_info_f.npz.
> > > Since the `tfs_c_inv ` we used here is to align the rest pose of CLOTH3D to the rest pose of SMPL.
> > > ```
> > > data_body = np.load(os.path.join(extra_dir, 'body_info_f.npz'))
> > > tfs_c_inv = torch.FloatTensor(data_body['tfs_c_inv']).to(device)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The rest pose of CLOTH3D is defined as below instead of a zero vector as that defined in SMPL.
> > > ```
> > > pose = pose.reshape(1, 24, 3)
> > > pose[0, 1, 2] = 0.15
> > > pose[0, 2, 2] = -0.15
> > > pose = pose.reshape(1, 72)
> > > ```
> > 
> > 
> > I have an .obj file containing posed SMPL, how can i use that for draping. As i checked it out u have taken the pose component from sample-pose.pt file and i wanned to know can i extract that pose parameters from my posed_smpl.obj file
> 
> Yes, you can. You need to fit smpl model to your obj mesh to get the corresponding \theta and \beta.

Can you please provide mode resources on how to do it
